Add Solver enum

hongkai-dai opened the pull request:
`SolverName()` becomes a static method, and add a enum to represent each solver. This is more robust than comparing strings representing solver name.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/robotlocomotion/drake/5021)
<!-- Reviewable:end -->

hongkai-dai commented:
+@soonho-tri for feature review.

---

Review status: 0 of 26 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-:-Kbw_8KymDoTlBSk-9FP:bboswjz)*
<!-- Sent from Reviewable.io -->

soonho-tri commented:
Reviewed 1 of 26 files at r1.
Review status: 1 of 26 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[drake/solvers/mathematical_program_solver_interface.h, line 21 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-KbzS32N--QuaxqpM85n:-KbzS32O6JOT8trLXcnU:b-hjnbc6) ([raw file](https://github.com/robotlocomotion/drake/blob/11360fd9f7a1822725533e945c38ae463b1055b8/drake/solvers/mathematical_program_solver_interface.h#L21)):*
> ```objc
> class MathematicalProgramSolverInterface {
>  public:
>   enum Solver {
> ```

I think it's better to use enum class here.
The following link includes more information about it: http://stackoverflow.com/questions/18335861/why-is-enum-class-preferred-over-plain-enum

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5021)*
<!-- Sent from Reviewable.io -->

soonho-tri commented:
Review status: 1 of 26 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-KbzSbi1szyPyTli0IdL:-KbzSbi2A6jIGXv5sZcX:b-qax6fk) (no related file):*
There are two bazel build failures on CI. I'm testing it on my local machine...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5021)*
<!-- Sent from Reviewable.io -->

soonho-tri commented:
Review status: 1 of 26 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-KbzSbi1szyPyTli0IdL:-KbzXMWT6uemqK3LZv16:bnfft1w) (no related file):*
<details><summary><i>Previously, soonho-tri (Soonho Kong) wrote…</i></summary><blockquote>

Please update the `drake/solvers/BUILD` file as follows:
```
diff --git a/drake/solvers/BUILD b/drake/solvers/BUILD
index ba45e81d9..b5e83fed4 100644
--- a/drake/solvers/BUILD
+++ b/drake/solvers/BUILD
@@ -70,7 +70,6 @@
     srcs = [
         "dreal_solver.h",
         "equality_constrained_qp_solver.cc",
-        "equality_constrained_qp_solver.h",
         "linear_system_solver.cc",
         "linear_system_solver.h",
         "mathematical_program.cc",
@@ -85,6 +84,7 @@
         "snopt_solver.h",
     ],
     hdrs = [
+        "equality_constrained_qp_solver.h",
         "mathematical_program.h",
         "mathematical_program_solver_interface.h",
     ],
@@ -239,7 +239,7 @@
         "test/mathematical_program_test_util.h",
     ],
     deps = [
-        "mathematical_program",
+        ":mathematical_program",
         "//drake/common:eigen_matrix_compare",
     ],
 )
```
</blockquote></details>

@david-german-tri , Hongkai has a test which includes `equality_constrained_qp_solver.h` header. Could you confirm that the above is the right solution in this context?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5021)*
<!-- Sent from Reviewable.io -->

soonho-tri commented:
Reviewed 25 of 26 files at r1.
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[drake/solvers/linear_system_solver.cc, line 50 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-Kbz_arSE_niwJz9h3gJ:-Kbz_arSE_niwJz9h3gK:b9t0vos) ([raw file](https://github.com/robotlocomotion/drake/blob/11360fd9f7a1822725533e945c38ae463b1055b8/drake/solvers/linear_system_solver.cc#L50)):*
> ```cpp
>       Aeq.jacobiSvd(Eigen::ComputeThinU | Eigen::ComputeThinV).solve(beq));
> 
>   prog.SetSolverResult(LinearSystemSolver::SolverName(), 0);
> ```

META: `LinearSystemSolver::` part is not necessary.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5021)*
<!-- Sent from Reviewable.io -->

soonho-tri commented:
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-KbzaUPgTk86tpdPpkI9:-KbzaUPgTk86tpdPpkIA:bwq1jml) (no related file):*
We still have `std::string solver_name_` member in `MathematicalProgram` class, and `SetSolverOption`/`GetSolverResult`/`SetSolverResult` are taking `solver_name` parameter. Is it possible to change them to use `solver_type`? 

---

*[a discussion](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-Kbzb-w-NQcwKJL-DoaF:-Kbzb-w-NQcwKJL-DoaG:b5bz52l) (no related file):*
A solver implementing `MathematicalProgramSolverInterface` have two methods:

 - `Solver solver_type()`
 - `std::string SolverName()`

For now, the two methods can be inconsistent (by a mistake). I think the following can improve the implementation. What do you think?

 1. Introduce a member of `Solver` to `MathematicalProgramSolverInterface` class (+ constructor taking `Solver`)
 2. `solver_type()` method simply returns `Solver` member.
 3. Implement `std::ostream& operator<<(std::ostream& os, const Solver& solver_type)` which outputs solver_name. It could be useful in debugging as well.
 4. `SolverName()` method use `solver_type` and `operator<<` (+ `ostringstream`).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5021)*
<!-- Sent from Reviewable.io -->

david-german-tri commented:
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-KbzSbi1szyPyTli0IdL:-Kbzf17I-kMlVl5HvIBx:b-cluc40) (no related file):*
> Hongkai has a test which includes equality_constrained_qp_solver.h header. Could you confirm that the above is the right solution in this context?

Looks correct to me.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5021)*
<!-- Sent from Reviewable.io -->

soonho-tri commented:
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-KbzSbi1szyPyTli0IdL:-KbzgaGdBx9p8REwgynp:b-k8xing) (no related file):*
<details><summary><i>Previously, david-german-tri (David German) wrote…</i></summary><blockquote>

> Hongkai has a test which includes equality_constrained_qp_solver.h header. Could you confirm that the above is the right solution in this context?

Looks correct to me.
</blockquote></details>

@david-german-tri , Thank you!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5021)*
<!-- Sent from Reviewable.io -->

hongkai-dai commented:
Review status: 4 of 25 files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-KbzaUPgTk86tpdPpkI9:-Kc-0edV4bSEiq2DG9S3:byfof4z) (no related file):*
<details><summary><i>Previously, soonho-tri (Soonho Kong) wrote…</i></summary><blockquote>

We still have `std::string solver_name_` member in `MathematicalProgram` class, and `SetSolverOption`/`GetSolverResult`/`SetSolverResult` are taking `solver_name` parameter. Is it possible to change them to use `solver_type`? 
</blockquote></details>

I left `solver_name_` there as I think it provides debugging information. When the user calls `GetSolverResult`, he might want to print out the solver name to understand which solver is called. Returning a `solver_type` is not as understandable as returning a `solver_name`.

`SetSolverOption` actually accepts `MathematicalProgramSolverInterface::Solver` as the input, I will change the variable name to `solver_type`.

---

*[a discussion](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-Kbzb-w-NQcwKJL-DoaF:-Kc-u2nw8Ntk33-8jL-g:b-nzr8xo) (no related file):*
<details><summary><i>Previously, soonho-tri (Soonho Kong) wrote…</i></summary><blockquote>

A solver implementing `MathematicalProgramSolverInterface` have two methods:

 - `Solver solver_type()`
 - `std::string SolverName()`

For now, the two methods can be inconsistent (by a mistake). I think the following can improve the implementation. What do you think?

 1. Introduce a member of `Solver` to `MathematicalProgramSolverInterface` class (+ constructor taking `Solver`)
 2. `solver_type()` method simply returns `Solver` member.
 3. Implement `std::ostream& operator<<(std::ostream& os, const Solver& solver_type)` which outputs solver_name. It could be useful in debugging as well.
 4. `SolverName()` method use `solver_type` and `operator<<` (+ `ostringstream`).

</blockquote></details>

That is a nice design, thanks!

---

*[drake/solvers/mathematical_program_solver_interface.h, line 21 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-KbzS32N--QuaxqpM85n:-Kc-5TB5shLw3FrvfZdY:b-31am50) ([raw file](https://github.com/robotlocomotion/drake/blob/11360fd9f7a1822725533e945c38ae463b1055b8/drake/solvers/mathematical_program_solver_interface.h#L21)):*
<details><summary><i>Previously, soonho-tri (Soonho Kong) wrote…</i></summary><blockquote>

I think it's better to use enum class here.
The following link includes more information about it: http://stackoverflow.com/questions/18335861/why-is-enum-class-preferred-over-plain-enum
</blockquote></details>

Done. Good to know this, thanks!

---

*[drake/solvers/test/equality_constrained_qp_solver_test.cc, line 4 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-KbzXBGFH-GS8aSq2Ily:-Kc-u88CdD6kpAo6Reua:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/11360fd9f7a1822725533e945c38ae463b1055b8/drake/solvers/test/equality_constrained_qp_solver_test.cc#L4)):*
<details><summary><i>Previously, soonho-tri (Soonho Kong) wrote…</i></summary><blockquote>

should be `#include "drake/solvers/equality_constrained_qp_solver.h"`
</blockquote></details>

Done.

---

*[drake/solvers/linear_system_solver.cc, line 50 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-Kbz_arSE_niwJz9h3gJ:-Kc-u9a3-bjFZvMeMtMk:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/11360fd9f7a1822725533e945c38ae463b1055b8/drake/solvers/linear_system_solver.cc#L50)):*
<details><summary><i>Previously, soonho-tri (Soonho Kong) wrote…</i></summary><blockquote>

META: `LinearSystemSolver::` part is not necessary.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5021)*
<!-- Sent from Reviewable.io -->

soonho-tri commented:
Reviewed 23 of 25 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[drake/solvers/mathematical_program_solver_interface.cc, line 53 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-Kc07wDYRk1AHL6BxBmS:-Kc07wDYRk1AHL6BxBmT:bvrabgu) ([raw file](https://github.com/robotlocomotion/drake/blob/8367804032b2465eebfe7512fcb9e1d06c093707/drake/solvers/mathematical_program_solver_interface.cc#L53)):*
> ```cpp
> }
> 
> std::string Name(MathematicalProgramSolverInterface::Solver solver_type) {
> ```

I think the following is simpler:

```c++
  std::ostringstream oss;
  oss << solver_type;
  return oss.str();
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5021)*
<!-- Sent from Reviewable.io -->

soonho-tri commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-:-Kc0CNnSn39TxAzao296:bnfp4nl)*
<!-- Sent from Reviewable.io -->

hongkai-dai commented:
+@ggould-tri for platform review.

---

Review status: 25 of 26 files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/solvers/mathematical_program_solver_interface.cc, line 53 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-Kc07wDYRk1AHL6BxBmS:-Kc0Hg-c67Z1_k91XTzf:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/8367804032b2465eebfe7512fcb9e1d06c093707/drake/solvers/mathematical_program_solver_interface.cc#L53)):*
<details><summary><i>Previously, soonho-tri (Soonho Kong) wrote…</i></summary><blockquote>

I think the following is simpler:

```c++
  std::ostringstream oss;
  oss << solver_type;
  return oss.str();
```
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-:-Kc0Hjof3jhAXUxFOa_J:b-8qo47q)*
<!-- Sent from Reviewable.io -->

soonho-tri commented:
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5021)*
<!-- Sent from Reviewable.io -->

soonho-tri commented:
@drake-jenkins-bot linux-gcc-experimental-ros please

hongkai-dai commented:
@soonho-tri you are right, it is less error prone to store the `MathematicalProgramSolverInterface::Solver solver_type_` inside `MathematicalProgram`, instead of storing `std::string solver_name_`. I made the change.

---

Review status: 17 of 30 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-:-Kc0Tc5HUTbn30Cp2qC4:bul5kn3)*
<!-- Sent from Reviewable.io -->

soonho-tri commented:
Reviewed 1 of 13 files at r4.
Review status: 18 of 30 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5021)*
<!-- Sent from Reviewable.io -->

soonho-tri commented:
Great!

---

Review status: 18 of 30 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-:-Kc0VOTdUrnSKtSXgdke:btojbfd)*
<!-- Sent from Reviewable.io -->

soonho-tri commented:
Reviewed 12 of 13 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[drake/solvers/ipopt_solver.cc, line 383 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-Kc0mDqD6gd-DpBB7nJs:-Kc0mDqD6gd-DpBB7nJt:b-83sjoh) ([raw file](https://github.com/robotlocomotion/drake/blob/41286ea5cb84f48080e8c55e74a929f714e5937e/drake/solvers/ipopt_solver.cc#L383)):*
> ```cpp
>                                  IpoptCalculatedQuantities* ip_cq) {
>     problem_->SetSolverResult(
>         MathematicalProgramSolverInterface::Solver::kIpopt, status);
> ```

Nice!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5021)*
<!-- Sent from Reviewable.io -->

soonho-tri commented:
@drake-jenkins-bot linux-gcc-experimental-ros please

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-:-Kc0mxiJ2tHX1Fg0eddk:b-3f2dwx)*
<!-- Sent from Reviewable.io -->

ggould-tri commented:
+(status: curate commits before merging)
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> pending one blocking item below (also there are some optional suggestions).

---

Reviewed 2 of 26 files at r1, 14 of 25 files at r2, 1 of 1 files at r3, 12 of 13 files at r4.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[drake/multibody/inverse_kinematics_backend.cc, line 42 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-Kc3_hWOcp5ToRTku5-H:-Kc3_hWOcp5ToRTku5-I:b-vyo2hi) ([raw file](https://github.com/robotlocomotion/drake/blob/41286ea5cb84f48080e8c55e74a929f714e5937e/drake/multibody/inverse_kinematics_backend.cc#L42)):*
> ```cpp
> namespace systems {
> namespace plants {
> ```

FYI, at author's discretion consider namespace aliasing `MathematicalProgramSolverInterface::Solver` to somethings shorter to reduce boilerplate below.

---

*[drake/solvers/moby_lcp_solver.cc, line 88 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-Kc3azjljkWU0wx3kg_0:-Kc3azjljkWU0wx3kg_1:b-la9fev) ([raw file](https://github.com/robotlocomotion/drake/blob/41286ea5cb84f48080e8c55e74a929f714e5937e/drake/solvers/moby_lcp_solver.cc#L88)):*
> ```cpp
> // Sole constructor
> MobyLCPSolver::MobyLCPSolver() : MathematicalProgramSolverInterface(
>     Solver::kMobyLCP), log_enabled_(false) {}
> ```

Per style guide https://google.github.io/styleguide/cppguide.html#Constructor_Initializer_Lists initializer lists that don't fit on one line must place a newline before the colon.

---

*[drake/solvers/test/convex_optimization_test.cc, line 14 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-Kc3bMclLl8AyGE5IJOp:-Kc3bMclLl8AyGE5IJOq:b92avtw) ([raw file](https://github.com/robotlocomotion/drake/blob/41286ea5cb84f48080e8c55e74a929f714e5937e/drake/solvers/test/convex_optimization_test.cc#L14)):*
> ```cpp
> namespace solvers {
> namespace test {
> namespace {
> ```

FYI, at author's discretion consider `namespace solver = solvers::MathematicalProgramSolverInterface::Solver` to reduce boilerplate and improve readability below.

---

*[drake/solvers/test/equality_constrained_qp_solver_test.cc, line 8 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-Kc3bXf0zh0Pbvl7YTgc:-Kc3bXf1IQbHXKQ2LKq-:b-jzxrgb) ([raw file](https://github.com/robotlocomotion/drake/blob/41286ea5cb84f48080e8c55e74a929f714e5937e/drake/solvers/test/equality_constrained_qp_solver_test.cc#L8)):*
> ```cpp
> #include "drake/solvers/mathematical_program.h"
> #include "drake/solvers/test/mathematical_program_test_util.h"
> ```

FYI another good place to consider namespace aliasing.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-:-Kc3_U8eFndcN9MPvpK_:b-2dzyok)*
<!-- Sent from Reviewable.io -->

soonho-tri commented:
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[drake/solvers/moby_lcp_solver.cc, line 88 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-Kc3azjljkWU0wx3kg_0:-Kc3cnmUePIQ0HGDNyVb:bao06y5) ([raw file](https://github.com/robotlocomotion/drake/blob/41286ea5cb84f48080e8c55e74a929f714e5937e/drake/solvers/moby_lcp_solver.cc#L88)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

Per style guide https://google.github.io/styleguide/cppguide.html#Constructor_Initializer_Lists initializer lists that don't fit on one line must place a newline before the colon.
</blockquote></details>

FWIW, running `clang-format` reformats this into:
```c++
MobyLCPSolver::MobyLCPSolver()
    : MathematicalProgramSolverInterface(Solver::kMobyLCP),
      log_enabled_(false) {}
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5021)*
<!-- Sent from Reviewable.io -->

ggould-tri commented:
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[drake/solvers/moby_lcp_solver.cc, line 88 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-Kc3azjljkWU0wx3kg_0:-Kc3d5JZY8ybx4iVCML3:b-3rwe6z) ([raw file](https://github.com/robotlocomotion/drake/blob/41286ea5cb84f48080e8c55e74a929f714e5937e/drake/solvers/moby_lcp_solver.cc#L88)):*
<details><summary><i>Previously, soonho-tri (Soonho Kong) wrote…</i></summary><blockquote>

FWIW, running `clang-format` reformats this into:
```c++
MobyLCPSolver::MobyLCPSolver()
    : MathematicalProgramSolverInterface(Solver::kMobyLCP),
      log_enabled_(false) {}
```
</blockquote></details>

That looks good!  I recommend obeying clang-format on this one!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5021)*
<!-- Sent from Reviewable.io -->

hongkai-dai commented:
Review status: 7 of 30 files reviewed at latest revision, 4 unresolved discussions.

---

*[drake/multibody/inverse_kinematics_backend.cc, line 42 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-Kc3_hWOcp5ToRTku5-H:-Kc4KGIITzeONip7eTpf:b-gtwcp4) ([raw file](https://github.com/robotlocomotion/drake/blob/41286ea5cb84f48080e8c55e74a929f714e5937e/drake/multibody/inverse_kinematics_backend.cc#L42)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

FYI, at author's discretion consider namespace aliasing `MathematicalProgramSolverInterface::Solver` to somethings shorter to reduce boilerplate below.
</blockquote></details>

Done. I moved `Solver` enum class out of `MathematicalProgramSolverInterface`, and renamed it as `SolverType`. I think this is better since I can also use shorter namespace in `.h` file.

---

*[drake/solvers/moby_lcp_solver.cc, line 88 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-Kc3azjljkWU0wx3kg_0:-Kc4B8e8VY4XtimshwJw:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/41286ea5cb84f48080e8c55e74a929f714e5937e/drake/solvers/moby_lcp_solver.cc#L88)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

That looks good!  I recommend obeying clang-format on this one!
</blockquote></details>

Done.

---

*[drake/solvers/test/convex_optimization_test.cc, line 14 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-Kc3bMclLl8AyGE5IJOp:-Kc4KQsEpSfYJXUEb4cq:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/41286ea5cb84f48080e8c55e74a929f714e5937e/drake/solvers/test/convex_optimization_test.cc#L14)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

FYI, at author's discretion consider `namespace solver = solvers::MathematicalProgramSolverInterface::Solver` to reduce boilerplate and improve readability below.
</blockquote></details>

Done.

---

*[drake/solvers/test/equality_constrained_qp_solver_test.cc, line 8 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-Kc3bXf0zh0Pbvl7YTgc:-Kc4KScdxVZ_FImqMUrv:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/41286ea5cb84f48080e8c55e74a929f714e5937e/drake/solvers/test/equality_constrained_qp_solver_test.cc#L8)):*
<details><summary><i>Previously, ggould-tri wrote…</i></summary><blockquote>

FYI another good place to consider namespace aliasing.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5021)*
<!-- Sent from Reviewable.io -->

ggould-tri commented:
:thumbsup: 

---

Reviewed 23 of 23 files at r5.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/multibody/inverse_kinematics_backend.cc, line 42 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-Kc3_hWOcp5ToRTku5-H:-Kc4Kw-ESBbtyiCJuLMc:b-rggmkg) ([raw file](https://github.com/robotlocomotion/drake/blob/41286ea5cb84f48080e8c55e74a929f714e5937e/drake/multibody/inverse_kinematics_backend.cc#L42)):*
<details><summary><i>Previously, hongkai-dai (Hongkai Dai) wrote…</i></summary><blockquote>

Done. I moved `Solver` enum class out of `MathematicalProgramSolverInterface`, and renamed it as `SolverType`. I think this is better since I can also use shorter namespace in `.h` file.
</blockquote></details>

Great thought!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-:-Kc4LRvdDYY8Qzpa3ljj:b4c9lyr)*
<!-- Sent from Reviewable.io -->

hongkai-dai commented:
-(status: curate commits before merging) 

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/5021#-:-Kc4VcQJE567uY081Xxk:b-l70wzf)*
<!-- Sent from Reviewable.io -->
